docs/troubleshooting.md

```diff
@@ -76,3 +76,17 @@ the query execution. We will evaluate if it's  worth also externalizing SPO and
 SOP permutations in this way to further reduce the RAM usage, or to let the user
 decide which permutations shall be stored in which format.
 
+## Note for Mac users.
+
+QLever preallocates a large (1TB) virtual file on disk for index. 
+This is done for performance reasons and 
+does not really allocate this amount on disk for standard Linux file systems (EXT3, etc). 
+However, this can cause an error on Mac APFS file system:
```

Comment:
I don't think "virtual file" is the right word here, also EXT3 has been outdated for a while so we should say EXT4.

Maybe:
```
## Note when using Docker on Mac or Windows

When building an index, QLever will create scratch space for on-disc sorting. This space is allocated as a large 1 TB sparse file. On standard Linux file systems such as Ext4 sparse files are optimized and this will only take as much disk space as is actually used.
With some systems such as Docker on macOS or when using unsupported file systems such as NTFS or APFS, this may lead to problems as these do not properly support sparse files.

One possible error may be the following:
```


```
While macOS including Docker on Mac is not supported there are some workarounds.
You can manually change the constant `static const size_t STXXL_DISK_SIZE_INDEX_BUILDER` in [file](../src/global/Constants.h) or you can try using a named volume instead of a path on the host.
```